tests/functional/testSubscription.js

```diff
@@ -0,0 +1,426 @@
+        subscription: {
+          value: {
+            _subscription_type: 'web',
+            created: 23677200,
+            current_period_end: 2147483647,
+            cancel_at_period_end: true,
+            status: 'foo'
+          },
+          expected: {
+            activated: '10/2/70',
+            cancelled: '1/19/38',
+            label: 'Expires',
+            status: 'Inactive'
+          }
+        }
+      },
+      {
+        name: 'web subscription: active',
+        subscription: {
+          value: {
+            _subscription_type: 'web',
+            created: 23677200,
+            current_period_end: 2147483647,
+            cancel_at_period_end: true,
+            status: 'active'
+          },
+          expected: {
+            activated: '10/2/70',
+            cancelled: '1/19/38',
+            label: 'Expires',
+            status: 'Active'
+          }
+        }
+      },
+      {
+        name: 'web subscription: inactive',
+        subscription: {
+          value: {
+            _subscription_type: 'web',
+            created: 23677200,
+            current_period_end: 2147483647,
+            cancel_at_period_end: true,
+            status: 'inactive'
+          },
+          expected: {
+            activated: '10/2/70',
+            cancelled: '1/19/38',
+            label: 'Expires',
+            status: 'Inactive'
+          }
+        }
+      },
+      {
+        name: 'apple subscription: auto renew',
+        subscription: {
+          value: {
+            _subscription_type: 'iap_apple',
+            expiry_time_millis: 2147483647,
+            auto_renewing: true,
+          },
+          expected: {
+            cancelled: '1/25/70',
+            label: 'Next billed',
+            status: 'Inactive'
+          }
+        }
+      },
+      {
+        name: 'apple subscription: no auto renew',
+        subscription: {
+          value: {
+            _subscription_type: 'iap_apple',
+            expiry_time_millis: 2147483647,
+            auto_renewing: false,
+          },
+          expected: {
+            cancelled: '1/25/70',
+            label: 'Expires',
+            status: 'Inactive'
+          }
+        }
+      },
+      {
+        name: 'google subscription: auto renew',
+        subscription: {
+          value: {
+            _subscription_type: 'iap_google',
+            expiry_time_millis: 2147483647000,
+            auto_renewing: true,
+          },
+          expected: {
+            cancelled: '1/19/38',
+            label: 'Next billed',
+            status: 'Active'
+          }
+        }
+      },
+      {
+        name: 'google subscription: no auto renew',
+        subscription: {
+          value: {
+            _subscription_type: 'iap_google',
+            expiry_time_millis: 2147483647000,
+            auto_renewing: false,
+          },
+          expected: {
+            cancelled: '1/19/38',
+            label: 'Expires',
+            status: 'Active'
```

Comment:
Because the tests should not know how the code works, can we have inactive/active for both google/apple? Basically 4 tests.